checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py

```diff
@@ -1,26 +1,21 @@
-from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.common.models.enums import CheckCategories
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.consts import ANY_VALUE
 
-class cloudwatchLogGroupRetention(BaseResourceCheck):
+
+class cloudwatchLogGroupRetention(BaseResourceValueCheck):
     def __init__(self):
         name = "Ensure cloudwatch log groups specify retention days"
         id = "CKV_AWS_66"
         supported_resource = ['AWS::Logs::LogGroup']
         categories = [CheckCategories.LOGGING]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resource)
 
-    def scan_resource_conf(self, conf):
-        """
-            Looks for retention days in cloudwatch log group :
-            https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html#cfn-cwl-loggroup-retentionindays
-        :param conf: AWS::Logs::LogGroup configuration
-        :return: <CheckResult>
-        """
+    def get_inspected_key(self):
+        return 'Properties/RetentionInDays'
 
-        if 'Properties' in conf.keys():
-            if 'RetentionInDays' in conf['Properties'].keys():
-                return CheckResult.PASSED
-        return CheckResult.FAILED
+    def get_expected_value(self):
+        return ANY_VALUE
```

Comment:
@arielkru We've implemented it differently in terraform. Can we keep the consistency?

Reply:
this is the terraform code:
not very different. @nimrodkor what do you mean?
```

class cloudwatchLogGroupRetention(BaseResourceValueCheck):
    def __init__(self):
        name = "Ensure cloudwatch log groups specify retention days"
        id = "CKV_AWS_66"
        supported_resource = ['aws_cloudwatch_log_group']
        categories = [CheckCategories.LOGGING]
        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resource)

    def get_inspected_key(self):
        return "retention_in_days"

    def get_expected_value(self):
        return ANY_VALUE
```

Reply:
I thought we had a limit there (less than X days). Guess I was wrong